api/protocol/lorawan/device.proto

```diff
@@ -53,8 +53,26 @@ message Device {
   int64  last_seen = 21;
 }
 
+// SetRegisterOnJoinMessage signals whether we expect on-join registration for an app
+message SetRegisterOnJoinMessage {
+  // App identifier
+  string app_id  = 1;
+  // Toggle on-join registration
+  bool   val     = 2;
+  // AppEui sets the AppEUI non-registered devices have to join with
+  bytes app_eui = 3 [(gogoproto.customtype) = "github.com/TheThingsNetwork/ttn/core/types.AppEUI"];
+  // AppKey sets the AppKey that the new devices will register with 
+  bytes app_key = 4 [(gogoproto.customtype) = "github.com/TheThingsNetwork/ttn/core/types.AppKey"];
+  // AccessKey to the account server
+  string access_key = 5;
+  // AccessKeyName for the account server key
+  string access_key_name = 6;
```

Comment:
Okay. For the moment, the handler uses the same message format (since it implements the same `DeviceManagerServer` interface), but what I added is that I remove the keys before sending them from the handler. In `core/handler/manager_server.go`, from line 106:
```go
brokerMessage := *in
brokerMessage.AccessKey = ""
brokerMessage.AppKey = &types.AppKey{}
[...]
_, err = h.deviceManager.SetRegisterOnJoin(ctx, &brokerMessage)
if err != nil {
	return nil, errors.New("Broker did not set on-join registration setting")
}
```